Routers/Routers/FileDataHandler.cs

```diff
@@ -0,0 +1,121 @@
+﻿namespace Routers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Security.Cryptography;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Threading.Tasks;
+    using Graph;
+
+    /// <summary>
+    /// Provides static methods for reading and writing graph data to/from files.
+    /// </summary>
+    public static class FileDataHandler
+    {
+        /// <summary>
+        /// Parses an undirected graph from the specified file.
+        /// </summary>
+        /// <param name="path">Path to the input file.</param>
+        /// <returns>Parsed UndirectedGraph object.</returns>
+        public static UndirectedGraph ParseFile(string path)
+        {
+            StreamReader sr = new StreamReader(path);
+            UndirectedGraph graph = new UndirectedGraph();
```

Comment:
Не надо писать тип дважды:
```suggestion
            StreamReader sr = new(path);
            UndirectedGraph graph = new();
```